ci-operator/templates/openshift/openshift-azure/build-azure-image.yaml

```diff
@@ -0,0 +1,313 @@
+      resources:
+        requests:
+          cpu: 1
+          memory: 300Mi
+        limits:
+          cpu: 3
+          memory: 4Gi
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-secrets-azure
+        mountPath: /usr/share/ansible/openshift-ansible/inventory/dynamic/injected
+      env:
+      - name: HOME
+        value: /usr/share/ansible/openshift-ansible
+      - name: AZURE_VM_SIZE
+        value: "${AZURE_VM_SIZE}"
+      - name: JOB_NAME_SAFE
+        value: "${JOB_NAME_SAFE}"
+      - name: JOB_NAME_HASH
+        value: "${JOB_NAME_HASH}"
+      - name: OPENSHIFT_RELEASE
+        value: "${OPENSHIFT_RELEASE}"
+      - name: TYPE
+        value: azure
+      - name: DEPLOY_OS
+        value: "${DEPLOY_OS}"
+      - name: INSTANCE_PREFIX
+        value: ${NAMESPACE}-${JOB_NAME_HASH}
+      command:
+      - /usr/local/bin/entrypoint-provider
+      args:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        # error handling and sync code
+        set -euo pipefail
+
+        trap '[ $? == 0 ] || touch /tmp/shared/exit' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        # We need to persist this variable so that we can clean it up when we exit
+        echo "RESOURCE_GROUP=${INSTANCE_PREFIX}-$(TZ=Etc/UTC date +%Y%m%d%H%M)" >>/tmp/shared/environment
+        echo "IMAGE_NAME=${DEPLOY_OS}-$OPENSHIFT_RELEASE-$(TZ=Etc/UTC date +%Y%m%d%H%M)" >>/tmp/shared/environment
+
+        # Get resource group and image name
+        source /tmp/shared/environment
+        echo "RESOURCE_GROUP=${RESOURCE_GROUP}"
+        echo "IMAGE_NAME=${IMAGE_NAME}"
+        echo $(pwd)
+
+        # Source azure creds
+        source /usr/share/ansible/openshift-ansible/inventory/dynamic/injected/credentials
+
+        ansible-playbook \
+          -e openshift_azure_resource_group_name=$RESOURCE_GROUP \
+          -e openshift_azure_resource_location=eastus \
+          -e openshift_azure_input_image_ns=images \
+          -e openshift_azure_input_image_prefix=${DEPLOY_OS}-base \
+          -e openshift_azure_output_image_ns=images \
+          -e openshift_azure_output_image_name=$IMAGE_NAME \
+          -e openshift_azure_storage_account=openshiftimages \
+          -e openshift_azure_storage_account_ns=images \
+          -e openshift_azure_container=images \
+          -e openshift_azure_vm_size=$AZURE_VM_SIZE \
+          -e openshift_azure_install_repo=$(curl -qs https://storage.googleapis.com/origin-ci-test/releases/openshift/origin/release-${OPENSHIFT_RELEASE}/.latest-rpms) \
+          playbooks/azure/openshift-cluster/build_node_image.yml
+
+        az login --service-principal -u ${AZURE_CLIENT_ID} -p ${AZURE_CLIENT_SECRET} --tenant ${AZURE_TENANT} &>/dev/null
+        az group delete -n ${RESOURCE_GROUP} -y
+
+        touch /tmp/shared/image_create
+
+    # Runs an install
+    - name: setup
+      image: registry.svc.ci.openshift.org/azure/openshift-azure:latest
```

Comment:
Webhook to our repo to rebuild on each PR merge?

Reply:
Follow-up issue. I am not crazy about rebuilding that on every commit that gets into master.

Reply:
Though we will likely want to do that.

Reply:
https://github.com/openshift/openshift-azure/issues/438